src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataPropertyFilter.cs

```diff
@@ -44,6 +49,19 @@ public void OnTempDataSaving(ITempDataDictionary tempData)
             }
         }
 
+        /// <summary>
+        /// Applies values from TempData from <paramref name="httpContext"/> to the <paramref name="subject"/>.
+        /// </summary>
+        /// <param name="subject">The properties of the subject are set based on TempData from
+        /// <paramref name="httpContext"/>. May be a <see cref="Controller"/>.</param>
+        /// <param name="httpContext">The <see cref="HttpContext"/> used to find TempData.</param>
+        public void ApplyTempDataChanges(object subject, HttpContext httpContext)
+        {
+            var tempData = _factory.GetTempData(httpContext);
+
+            SetPropertyVaules(tempData, subject);
```

Comment:
Can we set the `Subject` outside of this so the API is symmetrical for action filters and this call i.e.

```
filter.Subject = page;
filter.ApplyTempDataChanges(httpContext);
```